internal/controllers/core/cluster/reconciler.go

```diff
@@ -103,6 +104,10 @@ func (r *Reconciler) Reconcile(ctx context.Context, request reconcile.Request) (
 		}
 	}
 
+	if !hasConnection {
```

Comment:
Maybe rather than reporting on `if !hasConnection` you track a separate boolean `connectionChanged` and report on when that is true?